File: FizzBuzz.Backend/FizzBuzz.Backend.API/Entities/Game.cs
namespace FizzBuzz.Backend.API.Entities
{
    public class Game : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string Author { get; set; }= string.Empty;
        public int MinValue { get; set; } = 1;
        public int MaxValue { get; set; } = 100;
        public ICollection<GameRule> Rules { get; set; } = [];
    }
}

File: fizzbuzz.frontend/src/types/game.ts
export interface GameRule {
    divisor: number;
    word: string;
}

export interface Game {
    id: string;
    name: string;
    author: string;
    minValue: number;
    maxValue: number;
    createdAt: string;
    rules: GameRule[];
}

export interface CreateGameRequest {
    name: string;
    author: string;
    minValue: number;
    maxValue: number;
    rules: GameRule[];
}
